### PR TITLE
KV Store 

### PR DESCRIPTION
**Pull request by ThatGalNatalie**
PR Checklist:

- [x] add a short description of what's changed to the top of the `CHANGELOG.md`
- [ ] add/update tests (or don't, for reasons explained below)

## Describe this PR
Allow users to store small pieces of data relevant to their workflows

Screen Shot 2021-05-27 at 4 46 28 PM

### CODE REVIEW

File: src/pages/TeamSettings/KV.vue
Comment (zhen0) [inline src/pages/TeamSettings/KV.vue:114]: I get an error if I enter a number as a value. CodeMirror expects it to be input as a string.  Is there a reason we're only stringify-ing an object?

File: src/pages/TeamSettings/KV.vue
Comment (zhen0) [inline src/pages/TeamSettings/KV.vue:418]: Add a prepend icon

File: src/pages/TeamSettings/KV.vue
Comment (zhen0) [inline src/pages/TeamSettings/KV.vue:475]: I would switch item to type so that it's got a unique name.
```suggestion
                        v-for="(type, index) in kvTypes"
```

File: src/pages/TeamSettings/KV.vue
Comment (zhen0) [inline src/pages/TeamSettings/KV.vue:478]:
```suggestion
                        <v-list-item-title>{{ type.text }} </v-list-item-title>
```

**Review by zhen0 — DISMISSED** [review]
Lots of small changes @ThatGalNatalie. Let me know if you have any questions. 

Also - please check if you want to include the package-lock.json

File: src/components/Nav/TeamMenu.vue
Comment (zhen0) [inline src/components/Nav/TeamMenu.vue:169]:
```suggestion
              KV Store
```

File: src/pages/TeamSettings/KV.vue
Comment (zhen0) [inline src/pages/TeamSettings/KV.vue:347]:
```suggestion
        <ExternalLink href="https://docs.prefect.io/">key:value</ExternalLink> -->
```

**Review by zhen0 — COMMENTED** [review]
Two small suggestions/nitpicks from me.  

Also could we add a default sort e.g. by Key - I found the keys jumping around when I add a new one or edit one.  I think you can do that with the sort-by prop?

**Review by zhen0 — COMMENTED** [review]
This looks good to me but I'll let @znicholasbrown review before we merge.

**Review by znicholasbrown — DISMISSED** [review]
First thoughts:
- No error message is thrown when the `set_key_value` mutation fails; we need to show a helpful message in particular for instances where key count is exceeded but in general the loading state shouldn't persist if the mutation fails
- Searchbar styling should match the searchbars on other team management pages:
Screen Shot 2021-05-24 at 1 39 35 PM
- The team menu icon should use an FA-style icon so it matches the rest of the non-Prefect icons; [this one](https://fontawesome.com/icons/brackets-curly?style=duotone) or [this one](https://fontawesome.com/icons/database?style=duotone) might be good candidates

Still need to do some testing so will add another review later 👍🏻

**Comment by zhen0** [comment]
@znicholasbrown - This one is ready for your review also.

**Review by znicholasbrown — DISMISSED** [review]
Some more thoughts:
- It'd be good to have a count of the current number of keys out of how many the license allows, and then to disable the `Add KV` button when that count is at max
- I can't currently set a value to null, I'd expect to be able to do this unless there's an API limitation to this (this applies to both creating a new key and updating a key value)
- It doesn't look like the KV table is polling at all, I need to reload the page to get updated key values; some generous polling would be good here, maybe 30 seconds?
- I'm not sure the `Created by` column is adding much value, since that only reflects the current team and switching a team loads only the values for that team
- I don't think this info box is needed now that the panels are expandable (it's also broken for long pieces of data):
Screen Shot 2021-05-27 at 3 10 07 PM

- I don't think we need the placeholder text of the reset value like here (this confused me when I tried to clear the box):
Screen Shot 2021-05-27 at 3 11 09 PM


Some small notes on the create modal:
Screen Shot 2021-05-27 at 3 20 15 PM
1. This header can be "Create KV"
2. Can just be "Key"
3. Can just be "Value" ( no need for the type validation hint either )

I'd also switch text of the "Confirm" button to "Save"

**Review by znicholasbrown — DISMISSED** [review]
Thanks for the changes @ThatGalNatalie ! I've added a few comments re: loading states/race conditions. This is on a license without access to the feature:
Screen Shot 2021-06-01 at 10 19 46 AM

In a case like this I think it'd make the most sense to hide the table/add buttons and show something about contacting us to get access to the feature.

File: src/pages/TeamSettings/KV.vue
Comment (ThatGalNatalie) [reply src/pages/TeamSettings/KV.vue:198]: This line is necessary when updating but not when creating. Will add that in.

File: src/pages/TeamSettings/KV.vue
Comment (ThatGalNatalie) [reply src/pages/TeamSettings/KV.vue:343]: There is not a reason to not show that so I'll rm that

**Comment by zhen0** [comment]
@znicholasbrown - we'd love to get this merged today so that Zach's blog post can go out. Let me know if there's anything I can do to help speed up review.

**Review by znicholasbrown — APPROVED** [review]
LGTM, thanks for the changes @ThatGalNatalie 👍🏻

**Pull request merged**
